### PR TITLE
deprecate scalar inputs for any and all built-in functions

### LINKED ISSUE #178: Specify semantics of any() and all() for scalar inputs
The OpenCL C specification describes the `any()` and `all()` built-in functions as follows:

Function | Description
-- | --
int **any**(igentype _x_) | Returns 1 if the most significant bit in any component of _x_ is set; otherwise returns 0.
int **all**(igentype _x_) | Returns 1 if the most significant bit in all components of _x_ is set; otherwise returns 0.

This implies that the semantics of these functions have only been defined for vector types. What are the semantics for scalar types?

I would expect that `any(0) == false`, `all(0) == false`, `any(-1) == true` and `all(-1) == true)`, but it is not clear what `any(1)` and `all(1)` would evaluate to. Should these compile? (Scalar types are listed for _igentype_ in the second paragraph of _6.13.6 Relational Functions_.)

### PR DESCRIPTION
**Pull request by bashbaug**
This change deprecates scalar inputs for the **any** and **all** built-in functions.

Fixes #178.

### CODE REVIEW

File: scripts/gen_dictionaries.py [hunk 1]
```diff
@@ -182,7 +182,7 @@ def GetFooter():
             category = type.get('category')
             if category == 'struct':
                 name = type.get('name')
-                print('found type: ' +name)
+                #print('found type: ' +name)
```
Comment (StuartDBrady) [inline scripts/gen_dictionaries.py:185]: This change seems unrelated.

Comment (bashbaug) [reply scripts/gen_dictionaries.py:185]: Yes, this removes some debug output leftover from https://github.com/KhronosGroup/OpenCL-Docs/pull/298 that I noticed while making these changes.

Comment (StuartDBrady) [reply scripts/gen_dictionaries.py:185]: Thanks. I just wanted to check that this change was intentional.
